Doc/library/re.rst

```diff
@@ -938,9 +938,8 @@ form.
       >>> print('[%s]+' % re.escape(legal_chars))
       [abcdefghijklmnopqrstuvwxyz0123456789!\#\$%\&'\*\+\-\.\^_`\|\~:]+
 
-      >>> operators = ['+', '-', '*', '/', '**']
-      >>> print('|'.join(map(re.escape, sorted(operators, reverse=True))))
-      /|\-|\+|\*\*|\*
+      >>> print(' '.join(map(re.escape, string.punctuation)))
```

Comment:
I prefer to keep the existing example. It is closer to real world code and contains some non-trivial tricks which you can copy to your code even if you don't understand it.

All these examples are simplified extractions from the real code and demonstrate different cases of using `re.escape()`.

Reply:
I appreciate that, but the reverse sort distracts from what's being done to the operators list, and concatenating with pipes makes it harder to read, which makes it easier to miss. Since we don't expect alphabetical characters to be escaped, it could alternatively replace the first code example (where the characters are concatenated without spaces between them, which makes it harder to read)

We could probably go with:

```python

print(' '.join([re.escape(c) for c in string.punctuation])

```

if we judge list comprehensions may be more readable in this case. We already have a map example in the lot.